ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_1d_program_factory.cpp

```diff
@@ -22,6 +22,20 @@ using ttnn::operations::unary::UnaryWithParam;
 
 namespace reuse_mcast_1d_optimized_helpers {
 
+uint32_t get_preferred_noc(const uint32_t src_x, const uint32_t dst_x, const tt_metal::Device* device) {
+    /*
+        NOC0: Preferred +x -> +y
+        NOC1: Preferred -y -> -x
+    */
+    uint32_t MAX_X = device->grid_size().x;
+
+    // Get the wrapped distances
+    uint32_t dist_right = src_x < dst_x ? dst_x - src_x : MAX_X - src_x + dst_x;
+    uint32_t dist_left =  src_x < dst_x ? src_x + MAX_X - dst_x : src_x - dst_x;
+
+    return dist_right < dist_left ? 0 : 1;
+}
+
```

Comment:
might be good try it out with the intended core arrangement  (cores placed near the dram banks)